src/handler.ts

```diff
@@ -17,48 +17,32 @@ const handler = async (
     `\nRunning Service:\n '${SERVICE}'\n mode: ${NODE_ENV}\n stage: '${AWS_STAGE}'\n region: '${AWS_REGION}'\n 
   Send to smc: ${SEND_TO_SMC}\n`,
   );
-  if (SEND_TO_SMC != undefined && SEND_TO_SMC.toUpperCase() === 'TRUE') {
-    try {
-      logger.debug(`Function triggered with '${JSON.stringify(event)}'.`);
 
-      for (const record of event.Records) {
+  const batchItemFailures: SQSBatchItemFailure[] = [];
+
+  if (SEND_TO_SMC?.toUpperCase() === 'TRUE') {
+    logger.debug(`Function triggered with '${JSON.stringify(event)}'.`);
+
+    for (const record of event.Records) {
+      try {
         const dynamoDBEvent: DynamoDBRecord = JSON.parse(record.body) as DynamoDBRecord;
-        if (dynamoDBEvent){
-          const mcRequests: MCRequest[] = extractMCTestResults(dynamoDBEvent);
+        const mcRequests: MCRequest[] = extractMCTestResults(dynamoDBEvent);
 
-          if (mcRequests != null && mcRequests.length > 0) {
-            await sendMCProhibition(mcRequests);
-          } else {
-            logger.info(`No relevant MC test results found in the record: ${JSON.stringify(dynamoDBEvent)}`);
-          }
+        if (mcRequests.length > 0) {
+          await sendMCProhibition(mcRequests);
         } else {
-          logger.info('Function not triggered, empty notification.');
-          callback(null, 'Function not triggered, empty notification.');
+          logger.info(`No relevant MC test results found in the record: ${JSON.stringify(dynamoDBEvent)}`);
         }
-      }
-
-      callback(null, 'Data processed successfully.');
-    } catch (error) {
-      if (error.body) {
-        logger.error(JSON.stringify(error.body));
-        callback(
-          null,
-          `Data processed unsuccessfully: ${JSON.stringify(error.body)}`,
-        );
-      } else {
-        logger.error(error);
-        callback(null, `Data processed unsuccessfully: ${error}`);
+      } catch (error) {
+        logger.error(`Error processing record: ${JSON.stringify(record)}`);
+        batchItemFailures.push({ itemIdentifier: record.messageId });
       }
     }
   } else {
-    logger.info(
-      'Function not triggered, Missing or not true environment variable present',
-    );
-    callback(
-      null,
-      'Function not triggered, Missing or not true environment variable present',
-    );
+    logger.info('Function not triggered, Missing or not true environment variable present');
   }
+
+  callback(null, { batchItemFailures });
```

Comment:
can simplify this just down to be a return statement I think. Using callback is not needed and I think a bit old school these days

Reply:
I think it should be fine, you've changed that this function returns anyway already. Regression testing on this needs to include a successful run anyway so this should be covered already